fix: fix api inconsistencies

chenshoo opened the pull request.

ronkatz96 committed:
fix: byte array in warnings and errors

ronkatz96 committed:
fix: byte array in warnings and errors

ronkatz96 committed:
fix: byte array in warnings and errors

ronkatz96 committed:
fix: byte array in warnings and errors

ronkatz96 committed:
fix: byte array in irrelevant entities

ronkatz96 committed:
fix: byte array in irrelevant entities

ronkatz96 committed:
fix: byte array in irrelevant entities

chenshoo committed:
refactor: lines seperators

chenshoo committed:
fix: tests

chenshoo committed:
fix: tests

chenshoo committed:
fix: tests

chenshoo committed:
fix: tests

chenshoo committed:
fix: tests

ronkatz96 commented on packages/polaris-core/jest.config.js:
```diff
@@ -5,6 +5,6 @@ module.exports = {
     projects: ["<rootDir>", "../polaris-test"],
     testMatch: ['**/*.test.ts'],
-    testTimeout: 150000,
+    testTimeout: 1000000,
```
revert?

itaykl commented on packages/polaris-core/src/middlewares/snapshot-middleware.ts:
```diff
@@ -26,13 +26,16 @@ export class SnapshotMiddleware {
+                this.logger.debug('Snapshot middleware finished job', context);
+                return currentPage;
+            } catch (e) {
+                context.errors = context.errors ? [e, ...context.errors] : [e];
```
why not to simply push "e" to context.errors?

itaykl replied:
if context errors is undefined? I don't want to initialize it with an empty array, just so I could push to it

itaykl commented on packages/polaris-core/src/plugins/extensions/extensions-listener.ts:
```diff
@@ -27,14 +27,19 @@ export class ExtensionsListener implements GraphQLRequestListener<PolarisGraphQL
+            if (context.errors) {
+                const errors: string[] = [];
+                context.errors?.forEach((err) => errors.push(err.toString()));
+                context.returnedExtensions.errors = errors;
```
why did you used this complex cast into array?

itaykl replied:
cause errors on the context object are of type polarisError, and I want to convert it to string array, so I could push it to the metadata buffer

itaykl commented on packages/polaris-nest/jest.config.js:
```diff
@@ -4,7 +4,7 @@ module.exports = {
     testEnvironment: 'node',
     testMatch: ['**/*.test.ts'],
-    testTimeout: 15000000,
+    testTimeout: 1000000,
```
revert?

itaykl commented on packages/polaris-test/jest.config.js:
```diff
@@ -4,7 +4,7 @@ module.exports = {
     testEnvironment: 'node',
     testMatch: ['**/*.test.ts'],
-    testTimeout: 150000,
+    testTimeout: 1000000,
```
revert?

itaykl commented on packages/polaris-test/shared-resources/logger.ts:
```diff
@@ -1,7 +1,7 @@
 
 const loggerConfig: LoggerConfiguration = {
-    loggerLevel: LoggerLevel.WARN,
+    loggerLevel: LoggerLevel.INFO,
```
revert?

itaykl commented on packages/polaris-typeorm/jest.config.js:
```diff
@@ -3,7 +3,7 @@ module.exports = {
     clearMocks: true,
     moduleFileExtensions: ['ts', 'js'],
-    testTimeout: 15000000,
+    testTimeout: 1000000,
```
revert?

Commit:
Merge branch 'development' of https://github.com/Enigmatis/polaris-united into fix-failed-snapshot-response

 Conflicts:
	package-lock.json

Commit:
refactor: fix tests and update readme

chenshoo commented:
refactored snapResponse not to return pagesIds.
refactored globalDataVersion extension to be dataVersion
fixed empty data in pages to return as empty array instead of empty object
fixed return message on a page that doesn't exist to be 'Snapshot page with id ${id} not found' instead of empty object
fixed return message on a snapshot metadata that doesn't exist to be 'Snapshot metadata with id ${id} not found' instead of empty object
refactored errors, irrelevant entities and warnings to be of buffer type
fixed getData in pagination so it would fetch by startIndex and pageSize as it is supposed to do, and not by startIndex and endIndex

Commit:
refactor: fix tests and update readme

Commit:
refactor: fix tests and update readme

Commit:
refactor: fix tests and update readme

Commit:
refactor: fix tests and update readme

Commit:
refactor: fix tests and update readme

Commit:
refactor: fix tests and update readme
